feat: Update alerts for ope classes

schwesig opened the pull request:
Introducing alerts for the rhods-notebooks namespace. Just one simple alert to check functionality and connection with vault before going with the more complex ones.

part of https://github.com/OCP-on-NERC/BU-RHOAI/issues/2

schwesig committed:
feat: Update alerts for ope classes

Introducing alerts for the rhods-notebooks namespace. Just one simple to check functionality before going with the more complex ones.

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

larsks commented on cluster-scope/base/external-secrets.io/externalsecrets/open-cluster-management-observability-alertmanager-config/externalsecret.yaml:
```diff
@@ -95,6 +102,16 @@ spec:
+          - name: slack-notifications-prod-rhods-ope
+            slack_configs:
+            - channel: '#alerts-prod-rhods-ope'
+              api_url: "{{ .slack_api_url_prod-rhods-ope }}"
```
I think all of these references are going to be invalid because `-` isn't allowed in identifiers. Instead of `slack_api_url_prod-rhods-ope` you will need `slack_api_url_prod_rhods_ope` (change in this section and in the `data` section).

schwesig replied:
@larsks this is very awkward :-( a stupid typo, grr-argh, thanks for finding it :+1:

schwesig committed:
fix: Replace - by _ in identifiers

Fixing typo: underscore replaces dash; following the review from Lars.

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

larsks committed:
Simplify value extraction from vault secrets

This replaces the many individual entries in the `data` section with two
references in the `dataFrom` section and appropriate rewriting rules to
avoid key conflicts.
